Sample/src/androidTest/java/dev/testify/sample/compose/glide/GlideImageScreenshotTest.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 ndtp
+  *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package dev.testify.sample.compose.glide
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.ui.Modifier
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.idling.CountingIdlingResource
+import dev.testify.ComposableScreenshotRule
+import dev.testify.annotation.ScreenshotInstrumentation
+import dev.testify.sample.glide.GlideImageDemo
+import org.junit.Rule
+import org.junit.Test
+
+class GlideImageScreenshotTest {
+
+    @get:Rule
+    val rule = ComposableScreenshotRule()
+
+    @ScreenshotInstrumentation
+    @Test
+    fun glideImageDemo() {
+        val idlingResource = CountingIdlingResource("ImageLoadCompletionResource", true).apply {
+            increment()
+        }
+
+        rule
+            .setCompose {
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .wrapContentHeight()
+                ) {
+                    GlideImageDemo(onImageLoaded = { idlingResource.decrement() })
+                }
+                IdlingRegistry.getInstance().register(idlingResource)
```

Comment:
I don't know compose very well yet. Is it safe to run non composable code like this in a composable block?

Reply:
Oh, you know, I was so eager to get this working that I didn't consider the composition lifecycle. It's allowed to run non-compose code in a composable block, but it needs to be done in such a way that it safely handle recomposition.
I'm going to take this offline and look more closely at this.
My gut says that this will get hammered by recompositions and that would be bad.

Thanks for the questions @gantonious